grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -2404,7 +2404,7 @@ class StockMovementService {
 
         if ((stockMovement?.hasBeenReceived() || stockMovement?.hasBeenPartiallyReceived()) && isEditing) {
             return g.message(code: "stockMovement.cantEditReceived.message")
-        } else if (!isSameOrigin && isDepot && stockMovement?.isPending() && !stockMovement?.isElectronicType()) {
+        } else if ((!isSameOrigin && isDepot && stockMovement?.isPending() && !stockMovement?.isElectronicType()) || (!isDepot && !isSameDestination)) {
```

Comment:
This is getting hard to read. Is there an option to simplify this one or improve readability?

Reply:
It can be moved to another if to improve readability.